RealmWeaver/ModuleWeaver.cs

```diff
@@ -303,29 +301,31 @@ private void WeaveType(TypeDefinition type, Dictionary<string, Tuple<MethodRefer
 
             ReplaceGetter(prop, columnName, methodTable[typeId].Item1);
             ReplaceSetter(prop, backingField, columnName, methodTable[typeId].Item2, typeImplementsPropertyChanged, propChangedEventDefinition, propChangedFieldDefinition);
-        }
-//                else if (prop.PropertyType.Name == "IList`1" && prop.PropertyType.Namespace == "System.Collections.Generic")
-        else if (prop.PropertyType.Name == "RealmList`1" && prop.PropertyType.Namespace == "Realms")
+        } 
+
+        // treat IList and RealmList identically
+        else if ((prop.PropertyType.Name == "IList`1" && prop.PropertyType.Namespace == "System.Collections.Generic")  ||
+            (prop.PropertyType.Name == "RealmList`1" && prop.PropertyType.Namespace == "Realms"))
         {
-            // RealmList allows people to declare lists only of _realmObject due to the class definition
-            if (!prop.IsAutomatic())
-            {
-                LogErrorPoint(
-                    $"{type.Name}.{columnName} is not an automatic property but its type is a RealmList which normally indicates a relationship",
-                    sequencePoint);
+            var elementType = ((GenericInstanceType)prop.PropertyType).GenericArguments.Single();
+            // IList or RealmList allows people to declare lists only of _realmObject due to the class definition
+            if (!elementType.Resolve().BaseType.IsSameAs(_realmObject)) {
+                LogWarningPoint(
```

Comment:
I can't decide if I think this should be a warning or an error.
In favor of it being an error: you can add `[Ignore]` to explicitly state that you know what you are doing, but you might miss the warning. On the other hand, it's more in line with our decision to implicitly ignore properties with non-automatic getters or setters.